### ISSUE TITLE
ClaimsCheck throwing TypeError: Cannot read property 'claims' of undefined

### ISSUE THREAD

**Issue by thomasmburke** [FirebaseExtended/reactfire#325]
### Version info

**React:** ^16.14.0

**Firebase:** 8.2.1

**ReactFire:** 3.0.0-rc.0

### Steps to reproduce

Seems to happen with any `AuthCheck` that uses the `requiredClaims` param or by calling `ClaimsCheck` directly
```jsx
<AuthCheck fallback={<Home />} requiredClaims={{admin: true}}>
    <Admin />
</AuthCheck>}
```

### Expected behavior

Users with custom claims should pass the `AuthCheck` or `ClaimsCheck` and fallback to the fallback argument

### Actual behavior

Receiving error: `TypeError: Cannot read property 'claims' of undefined`

![image](https://user-images.githubusercontent.com/40719837/104142674-68d2fe00-5371-11eb-92d4-ae3e0ed7f1c5.png)

Debugging in Chrome Dev tools indicates that the `useObservable` call may have an unresolved promise? I'm guessing this because `_useIdTokenResult` has a status of "loading" and `isComplete` is false.

![image](https://user-images.githubusercontent.com/40719837/104142807-f57dbc00-5371-11eb-8f0a-0f95a2a5c7ee.png)

Additionally, when I expand the user variable that I have blacked out I can see that it successfully is getting my authenticated user, meaning I can see the email, displayName, and custom claims when inspecting the JWT token. The same `<AuthCheck>` call works if I omit the `requiredClaims` param.

**Comment by Ljugoboss** [comment]
I'm running into the same issue.

The problem seems to occur due to `latest` always ending up defaulting to `config.initialData` in `useObservable` the first time it returns a value, but no config is sent in from `CheckClaims`;
![image](https://user-images.githubusercontent.com/4201301/105459552-b7926900-5c8a-11eb-9a94-3ae3180acc29.png)
![image](https://user-images.githubusercontent.com/4201301/105459171-0f7ca000-5c8a-11eb-91a4-fda8c5e15e66.png)

I solved it locally by adding default values to `data` and `claims`, or by actually sending in a config with `initialData`.
![image](https://user-images.githubusercontent.com/4201301/105459435-89148e00-5c8a-11eb-970a-59b1333515d8.png)
![image](https://user-images.githubusercontent.com/4201301/105459820-2a9bdf80-5c8b-11eb-9734-68434dd68726.png)

**Comment by ElinOlundForsling** [comment]
Is there a workaround to this that doesn't involve changing the node_modules files if an update is upcoming? Suspense, preload user?

**Comment by jhuleatt** [comment]
Some context on our slow response here:

`ClaimsCheck` was originally written when reactfire was Suspense/Concurrent mode-only, before the big [optional Suspense change](https://github.com/FirebaseExtended/reactfire/pull/255). In the old version of ReactFire, `data` would never be `undefined` because the component wouldn't get to that line until the data fetch completed (thanks to Suspense).

We used to totally ignore loading states because Suspense would handle them. Making `AuthCheck` and `ClaimsCheck` support non-Suspense/Concurrent mode requires a bit more thought around how we want the api for these components and components like these to work going forward. 

For now, the easiest workaround is probably to copy the `AuthCheck` and/or `ClaimsCheck` [source](https://github.com/FirebaseExtended/reactfire/blob/main/src/auth.tsx) and modify it to your needs. That's basically what we did in the [withoutSuspense Auth example](https://github.com/FirebaseExtended/reactfire/blob/b4f22bc0a84729245db87861d5190a0483b19348/example/withoutSuspense/Auth.tsx#L10-L23).

In terms of how we may support this in the future, we'd love to hear thoughts from the community! One idea I've been thinking of is to make `AuthCheck` a hook instead. Something like this:

```jsx
function MyComponent() {
  const { data: isValidUser, status } = useAuthCheck({
    requiredClaims: { admin: true },
  });

  if (status === "loading") {
    return <LoadingSpinner />;
  }
  if (isValidUser === true) {
    return <AdminDashboard />;
  } else {
    return <YouDoNotHaveAccessToViewThisPage />;
  }
}
```

But definitely open to other ideas!
